### PR TITLE
[feature][transform-v2] Add FilterRowKindTransform to filter rows based on row kind

### PR DESCRIPTION
**Pull request by hailin0**
## Purpose of this pull request

Add FilterRowKindTransform to filter rows based on row kind

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Comment by hailin0** [comment]
@EricJoy2048 PTAL

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/FilterRowKindTransform.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    @Override
+    protected SeaTunnelRow transformRow(SeaTunnelRow inputRow) {
+        if (!excludeKinds.isEmpty()) {
+            return excludeKinds.contains(inputRow.getRowKind()) ? null : inputRow;
+        }
+        if (!includeKinds.isEmpty()) {
+            return includeKinds.contains(inputRow.getRowKind()) ? inputRow : null;
+        }
+        throw new SeaTunnelRuntimeException(CommonErrorCode.UNSUPPORTED_OPERATION,
+            "Transform config error! either excludeKinds or includeKinds must be configured");
```
Comment (TaoZex) [inline seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/FilterRowKindTransform.java]:
```suggestion
            "Transform config error! Either excludeKinds or includeKinds must be configured");
```

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/FilterRowKindTransform.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        if (pluginConfig.hasPath(INCLUDE_KINDS.key())) {
+            includeKinds = new HashSet<>(pluginConfig.getEnumList(RowKind.class, INCLUDE_KINDS.key()));
+        }
+        if (pluginConfig.hasPath(EXCLUDE_KINDS.key())) {
+            excludeKinds = new HashSet<>(pluginConfig.getEnumList(RowKind.class, EXCLUDE_KINDS.key()));
+        }
+        if ((includeKinds.isEmpty() && excludeKinds.isEmpty())
+            || (!includeKinds.isEmpty() && !excludeKinds.isEmpty())) {
+            throw new SeaTunnelRuntimeException(CommonErrorCode.ILLEGAL_ARGUMENT,
+                String.format("these options(%s,%s) are mutually exclusive, allowing only one set of options to be configured.",
```
Comment (TaoZex) [inline seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/FilterRowKindTransform.java]:
```suggestion
                String.format("These options(%s,%s) are mutually exclusive, allowing only one set of options to be configured.",
```

**Comment by hailin0** [comment]
image

**Review by TaoZex — DISMISSED** [review]
LGTM

**Review by ashulin — COMMENTED** [review]
@CalvinKirs Please help check the CI error

**Comment by CalvinKirs** [comment]
> @CalvinKirs Please help check the CI error↳

We need update code from upstream

**Comment by hailin0** [comment]
image

**Review by EricJoy2048 — APPROVED** [review]

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**